### PR TITLE
cbatticon: add module

### PR DESCRIPTION
**Pull request by pmiddend**
This adds a service for the `cbatticon` program. I included all command line options found in the manpage. This is my first home-manager PR, so I just hope everythings kind of okay :)

### CODE REVIEW

File: modules/services/cbatticon.nix [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+with lib;
+
+let
+
+  cfg = config.services.cbatticon;
+  package = pkgs.cbatticon;
+  commandLine = concatStringsSep " " (
+    optional
+      (cfg.commandCriticalLevel != null)
+      ("--command-critical-level '${cfg.commandCriticalLevel}'") ++
```
Comment (rycee) [inline modules/services/cbatticon.nix]: Perhaps the command should be wrapped in a shell script? I.e., having something like
```
let
  cmd = pkgs.writeShellScript "cbatticon-critical" cfg.commandCriticalLevel;
in
  "--command-critical-level '${cmd}'"
```
Similarly for `commandLeftClick`.

Then wouldn't have to worry about escaping the string and such.

**Comment by pmiddend** [comment]
@rycee Thanks for the thorough review. I've hopefully corrected everything you mentioned. :)

**Comment by pmiddend** [comment]
Anything blocking the merge here?

**Comment by rycee** [comment]
Sorry, no nothing in particular. I've just been a bit preoccupied elsewhere 🙂

Rebased to master in 3a3657b107bc1aa32619c41d00085bfb3f36d7c5 with a few minor edits. Thanks for the contribution!

**Pull request closed without merging**